### ISSUE TITLE
"Virtualbox is not using hardware-supported virtualization features"

### ISSUE THREAD

**Issue by kklmn** [myspaghetti/macos-virtualbox#651]
Hello,

I get the following complaint:

```
Checking that VirtualBox uses hardware-supported virtualization.

Virtualbox is not using hardware-supported virtualization features.
Check that software such as Hyper-V, Windows Sandbox, WSL2, memory integrity
protection, and other Windows features that lock virtualization are turned off.
Exiting.
```

Is there a way to tell which feature is to blame? Those listed above (not sure about memory integrity
protection) are unselected in my case:
![image](https://user-images.githubusercontent.com/17322580/230601217-2290fae3-e7ce-4928-a7d3-fcdf804ea46f.png)

**Comment by myspaghetti** [comment]
>Is there a way to tell which feature is to blame?

Unfortunately not, Windows doesn't provide a way to show which feature or software is using VT-x/AMD-V.

You can turn off all features that use Hyper-V virtualization by running the following command in an admin console then rebooting twice:

```
bcdedit /set hypervisorlaunchtype off 
```

**Comment by kklmn** [comment]
@myspaghetti, thank you for your reply!
Doing `bcdedit /set hypervisorlaunchtype off` (admin, rebooting twice) didn't help.

**Comment by myspaghetti** [comment]
Then either other software is preventing access to VT-x/AMD-V or the CPU does not have the required features.

**Comment by kklmn** [comment]
> ... or the CPU does not have the required features.

i7-12700H is the CPU

**Comment by RICCIARDI-Adrien** [comment]
@kklmn Is the virtualization support enabled in your BIOS ?

**Comment by kklmn** [comment]
Certainly yes, I do run a few other virtual machines.

**Comment by RICCIARDI-Adrien** [comment]
Maybe some application like Docker is keeping Hyper-V enabled ?
https://forums.virtualbox.org/viewtopic.php?f=6&t=105951

**Comment by kklmn** [comment]
Thanks for the hint, but I don't have Docker running and I have 3 other VMs in VirtualBox. I've checked now, I can start them normally.

**Comment by RICCIARDI-Adrien** [comment]
Do your working VMs use hardware virtualization (if you look at the Virtual Box system settings for such VMs) ?

**Comment by kklmn** [comment]
These are the settings:
![image](https://user-images.githubusercontent.com/17322580/231161009-17697105-af91-4aca-9355-ae6b07a0b4e0.png)
![image](https://user-images.githubusercontent.com/17322580/231161126-79a41291-b9f1-4af5-930a-16fd826e5c6b.png)
![image](https://user-images.githubusercontent.com/17322580/231161165-7f514f0e-9de9-4ad5-b1de-f0ed0ef50fc9.png)

**Comment by RICCIARDI-Adrien** [comment]
I'm using a 8th gen Intel CPU and I'm able to tick the "Enable Nested VT-x/AMD-V" checkbox (I'm running Linux as host OS).

![Capture d’écran_2023-04-11_14-38-10](https://user-images.githubusercontent.com/4292857/231164731-ab5c3be7-0db1-4800-9ae0-1586d59171b5.png)

I think this should be also available to you on your 12th gen CPU.

The selected paravirtualization interface is displayed in the VM description (see screenshot below). Which one is selected on your Fedora VM ?

![Capture d’écran_2023-04-11_14-41-54](https://user-images.githubusercontent.com/4292857/231165676-a52594f0-1e71-4caa-a69d-c80bcc595ca8.png)

**Comment by RICCIARDI-Adrien** [comment]
Maybe you can use the same commands than `macos-virtualbox` script to find if hardware virtualization is also supported on your other VMs ? A command like `VBoxManage showvminfo "Fedora 37" --log 0 | grep "Attempting fall back to NEM"` (borrowed from https://github.com/myspaghetti/macos-virtualbox/blob/master/macos-guest-virtualbox.sh#L463).

**Comment by kklmn** [comment]
I've got it!
First, I've found a great guide ["How to Disable or Remove Hyper-V in Windows 11"](https://www.makeuseof.com/windows-11-disable-hyper-v/) and followed it.
In my case, I needed to turn off Memory Integrity ("How to Turn Off Virtualization-Based Security (Memory Integrity)").
This alone didn't help. But then doing `bcdedit /set hypervisorlaunchtype off` as admin, double restart, did the trick.
Thanks all!
